jobserver/management/commands/release.py

```diff
@@ -1,26 +1,114 @@
 
 
 class Command(BaseCommand):
     help = "Release a directory of files to a workspace"  # noqa: A003
 
     def add_arguments(self, parser):
-        parser.add_argument("workspace_name", help="worksapace name to release to")
-        parser.add_argument("backend_name", help="backend to release from")
-        parser.add_argument("directory", help="directory of files to release")
+        parser.add_argument(
+            "directory", help="directory of files to release", type=Path
+        )
+        parser.add_argument(
+            "--workspace",
+            "-w",
+            dest="workspace_name",
+            default="test-workspace",
+            help="workspace name to release to",
+        )
+        parser.add_argument(
+            "--backend",
+            "-b",
+            dest="backend_name",
+            default="test-backend",
+            help="backend to release from",
+        )
+        parser.add_argument(
+            "--user",
+            "-u",
+            dest="username",
+            default="test-user",
+            help="user to release with",
+        )
+        parser.add_argument(
+            "--create",
+            "-c",
+            action="store_true",
+            help="create test workspace/backend/user if they do not exist",
+        )
+
+    def handle(
+        self,
+        directory,
+        workspace_name,
+        backend_name,
+        username,
+        create,
+        *args,
+        **options
+    ):
+
+        assert directory.exists()
+
+        files = {}
+
+        for filename in directory.glob("**/*"):
+            if filename.is_dir():
+                continue
+            relative = filename.relative_to(directory)
+            files[str(relative)] = hashlib.sha256(filename.read_bytes()).hexdigest()
 
-    def handle(self, workspace_name, backend_name, directory, *args, **options):
         try:
             workspace = Workspace.objects.get(name=workspace_name)
-            backend = Backend.objects.get(name=backend_name)
-            release_hash, zipstream = create_upload_zip(Path(directory))
-            handle_release(workspace, backend, "local user", release_hash, zipstream)
-        except Exception as e:
-            print(str(e), file=sys.stderr)
-            sys.exit(1)
+        except Workspace.DoesNotExist:
+            if create:
+                org = get_or_maybe_create(
+                    create, Org, {"name": "test-org"}, slug="test-org"
+                )
+                project = get_or_maybe_create(
+                    create,
+                    Project,
+                    {"name": "test-project"},
+                    slug="test-project",
+                    org=org,
+                )
+                workspace = Workspace.objects.create(
+                    name=workspace_name,
+                    project=project,
+                    repo="https://github.com/opensafely/test-repo",
+                )
+            else:
+                raise
```

Comment:
Seeing as this is such a short else block can we flip the conditional and not bother with an else? eg 

```
if not create:
    raise

…
```

Reply:
done